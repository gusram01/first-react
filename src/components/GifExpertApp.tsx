import React, { useState } from 'react';
import AddCategory from './AddCategory';
import GifGrid from './GifGrid';
import PropTypes from 'prop-types';

interface Props {
  defaultCategories?: string[];
}

const GifExpertApp: React.FC<Props> = ({ defaultCategories = [] }) => {

  const [categories, setCategories] = useState(defaultCategories);
  // const handleAdd = () => {
  //   setCategories([...categories].concat('SaintSeiya'));
  // }


  return (
    <>
      <h2>GifExpertApp</h2>
      <AddCategory setCategories={setCategories} />
      <hr />

      {/* <button onClick={handleAdd}>Agregar</button> */}

      <ol>
        {
          categories.map(category =>
            <GifGrid
              key={category}
              category={category}
            />)
        }
      </ol>
    </>
  )
}

GifExpertApp.propTypes = {
  defaultCategories: PropTypes.array
}

export default GifExpertApp;