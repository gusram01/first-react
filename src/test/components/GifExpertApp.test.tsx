import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import GifExpertApp from '../../components/GifExpertApp';


describe('GifExpertApp tests', () => {
  let wrapper: ShallowWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;

  test('Should render GifExpertApp correctly', () => {
    wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Should render categories array', () => {

    const categories = ['Dragon Ball', 'One Punch'];
    wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);

  });

});