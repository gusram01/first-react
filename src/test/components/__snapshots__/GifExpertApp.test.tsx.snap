// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp tests Should render GifExpertApp correctly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`GifExpertApp tests Should render categories array 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
  </ol>
</Fragment>
`;
